### DISCUSSION TITLE
type field in `exportJSON` for custom nodes

### DISCUSSION THREAD

**Discussion by IsmailElShinnawy** [facebook/lexical#3652] [Q&A]
Hi, I have been trying to implement an `exportJSON` method for `CustomHeadingNode` which extends `HeadingNode`, the problem is that `SerializedHeadingNode` type enforces that the type of the attribute `type` to be `"heading"`, but I want the custom `exportJSON` to return `type: "custom_heading"`.
As a temporary solution for now I typecast the returned json to `SerializedHeadingNode`, is there a cleaner approach for this?

Sandbox: https://codesandbox.io/s/flamboyant-brown-ojhvtq

**Comment by moy2010** [comment]
The cleaner way would be to either create a PR so that the `exportJSON` method in `HeadingNode` doesn't return `SerializedHeadingNode` but `SerializedElementNode`. Otherwise you can extend `ElementNode` instead of `HeadingNode` so that you don't have such type bound.
